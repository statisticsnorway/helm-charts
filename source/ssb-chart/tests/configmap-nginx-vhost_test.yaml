suite: test configmap-nginx-vhost
templates:
  - configmap-nginx-vhost.yaml
tests:
  
  - it: should render nothing if not enabled
    values: 
      - ./values/default-values.yaml
    asserts:
      - hasDocuments:
          count: 0
          
  - it: should render one document if enabled
    values: 
      - ./values/default-values.yaml
    set:
      backends:
        - name: "unittest-be"
          namespace: "unittesting"    
    asserts:
      - hasDocuments:
          count: 1

  - it: should include default labels, be of kind ConfigMap and generate name from app-name
    values: 
       - ./values/configmap-nginx-vhost-values.yaml
    asserts:
      - equal:
          path: metadata.labels
          value:
            app: unittest-app
            billing-project: ssb-unittest
            chart: ssb-chart-1.3.1
            heritage: Tiller
            release: RELEASE-NAME
      - isKind:
         of: ConfigMap
      - equal:
          path: metadata.name
          value: unittest-app-nginx-vhost-config             
  
  - it: should create vhost.conf     
    values: 
       - ./values/configmap-nginx-vhost-values.yaml
    asserts: 
      - matchSnapshot:
          path: data
