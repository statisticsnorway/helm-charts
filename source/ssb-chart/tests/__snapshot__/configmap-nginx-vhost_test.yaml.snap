should create vhost.conf:
  1: |
    vhost.conf: "# Upstream API's\nupstream unittest-be {\n  server unittest-be.unittesting;\n}\n\n#
      Upstream application\nupstream unittest-app {\n  server 127.0.0.1:80;\n}\n\n# Enable
      debug logging\nerror_log  \"/opt/bitnami/nginx/logs/error.log\" debug;\n\nserver
      {\n  listen 0.0.0.0:;\n  server_name _;\n\n  # Set proxy buffer sizes\n  proxy_buffering
      on;\n  proxy_buffer_size        128k;\n  proxy_buffers            4 256k;\n  proxy_busy_buffers_size
      \ 256k;\n\n  # Set header buffer sizes\n  client_header_buffer_size 32k;\n  large_client_header_buffers
      4 32k;\n\n  # API definitions\n  location /be/unittest-be {\n    set $upstream unittest-be;\n
      \   set $service unittest-be.unittesting;\n\n    rewrite ^/be/(.*)$ /_be/$1 last;\n
      \ }\n  \n  # API policy section\n  location /_be {\n    internal;\n\n    # Remove
      the prefixes\n    rewrite ^/[^/]+/[^/]+/(.*)$ /$1 break;\n\n    # Do not send upstream
      errors to the client\n    proxy_intercept_errors on;\n\n    # HTTP version (default
      is 1.0)\n    proxy_http_version 1.1;\n\n    # The original host requested by the
      client\n    proxy_set_header X-Forwarded-Host $host;\n\n    # Host header must be
      set to kubernetes \"<service_name.namespace>\"\n    # to be able to proxy the requests
      to the kubernetes domain\n    proxy_set_header Host $service;\n\n    # What other
      headers do we need?\n    proxy_set_header X-Real-IP $remote_addr;\n    proxy_set_header
      X-Scheme $scheme;\n\n    # Pass through only cookies with the name \"sessionToken\".\n
      \   # This is a Forbruk-team spesific setting and the parameters\n    # for \"proxy_set_header
      Cookie\" should maybe be made available\n    # as a variable in the Helmrelease
      in later improvements.\n    proxy_set_header Cookie $cookie_sessionToken;\n\n    #
      Pass request to upstream\n    proxy_pass http://$upstream;\n  }\n\n  # Nginx health
      check endpoint\n  location = /nginx-gw-health {\n    access_log off;\n    add_header
      Content-Type text/plain;\n\n    return 200 \"alive\\n\";\n  }\n\n  # Proxy everything
      else to the application.\n  location / {\n    proxy_http_version 1.1;\n    proxy_pass
      http://unittest-app;\n  }\n}"
